devel/yarn/Portfile

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem 1.0
+
+name yarn
+version 0.17.3
+
+categories          devel
+
+platforms           darwin
+supported_archs     noarch
+
+license             BSD
+
+maintainers         andrewsforge.com:code
+
+description         JavaScript dependency manager
+long_description    Created by Facebook to replace NPM, yarn boasts about its \
+                    speed, security, and reliability. Most notably, yarn \
+                    dependencies are installed deterministically, whereas \
+                    NPM's dependencies are not.
+
+homepage            https://yarnpkg.com/
+
+master_sites        https://yarnpkg.com/downloads/${version}/
+distname            ${name}-v${version}
+worksrcdir          dist
+checksums           rmd160  be20072bfd66a6411911bebb63284abb7f7f454c \
+                    sha256  883df435f68ce47c93c2e4be27acbd4122ae52ef4d334f75104c0c5e187a9173
+
+depends_run         bin:node:nodejs6
+variant nodejs4 conflicts nodejs6 nodejs7 description {Use Node.js v4} {
+    depends_lib-replace bin:node:nodejs6 port:nodejs4
+}
+variant nodejs6 conflicts nodejs4 nodejs7 description {Use Node.js v6} {
+    depends_lib-replace bin:node:nodejs6 port:nodejs6
+}
+variant nodejs7 conflicts nodejs4 nodejs6 description {Use Node.js v7} {
+    depends_lib-replace bin:node:nodejs6 port:nodejs7
+}
+
+post-patch {
+    reinplace -E "s;basedir=\\$\\(.+\\);basedir=\$(dirname \"\$(readlink \"\$0\" || echo \"\$(echo \"\$0\" | sed -e 's,\\\\\\\\,/,g')\")\");" ${worksrcpath}/bin/yarn
+}
+
+use_configure       no
+build {}
+
+destroot {
+  copy ${worksrcpath} ${destroot}${prefix}/libexec/yarn
+  ln -s ${prefix}/libexec/yarn/bin/yarn ${destroot}${prefix}/bin/yarn
+}
```

Comment:
Yarn is not a stand-alone executable. It's a shell script that invokes a JavaScript/Node script, which relies on other JavaScript files in the various directories around it.

I opted to put the scripts in `libexec` and to soft-link the shell script. We could put all the scripts and directories in `bin`, but that feels messy, especially because you don't want to call those scripts directly. We could alternatively put the scripts in `lib`, but no other program is going to rely on these scripts, so `libexec` seemed most appropriate.

Note that we could also soft-link the JavaScript script directly (instead of using the shell script) but I feel like I'm circumventing project code at that point.